packages/phoenix-evals/src/phoenix/evals/evaluators.py

```diff
@@ -111,9 +114,57 @@ class Score:
     label: Optional[str] = None
     explanation: Optional[str] = None
     metadata: Dict[str, Any] = field(default_factory=dict)
-    source: Optional[SourceType] = None
+    kind: Optional[KindType] = None
     direction: DirectionType = "maximize"
 
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        score: Optional[Union[float, int]] = None,
+        label: Optional[str] = None,
+        explanation: Optional[str] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+        source: Optional[SourceType] = None,
+        direction: DirectionType = "maximize",
+        *,
+        kind: Optional[KindType] = None,
+    ) -> None:
+        # Accept both 'kind' (preferred) and 'source' (deprecated)
+        if kind is not None and source is not None and kind != source:
+            raise ValueError("Provide only one of 'kind' or 'source' (they differ). Use 'kind'.")
+        resolved_kind: Optional[KindType] = kind if kind is not None else source
+        if source is not None and kind is None:
+            warnings.warn(
+                "Score 'source' is deprecated; use 'kind' instead.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+        # Heuristic deprecation
+        if resolved_kind == "heuristic":
+            warnings.warn(
+                "kind='heuristic' is deprecated; use kind='code' instead.",
```

Comment:
Should this warning mention that it is going to change the kind to "code" automatically as well?